pallets/dispatcher/src/lib.rs

```diff
@@ -169,6 +200,50 @@ pub mod pallet {
 			AaveManagerAccount::<T>::put(account);
 			Ok(())
 		}
+
+		/// Execute a single EVM call.
+		/// This extrinsic will fail if the EVM call reverts.
+		///
+		/// This can be used with pallet_utility::batch_all to achieve the same
+		/// functionality as batch_all itself with automatic revert on EVM call failure.
+		///
+		/// Parameters:
+		/// - `origin`: Signed origin.
+		/// - `target`: The EVM address of the contract to call.
+		/// - `input`: The input data for the call.
+		///
+		/// Emits `EvmCallSuccess` event when successful.
+		#[pallet::call_index(4)]
+		#[pallet::weight(T::GasWeightMapping::gas_to_weight(T::GasLimit::get(), false))]
+		pub fn dispatch_evm_call(
+			origin: OriginFor<T>,
+			target: EvmAddress,
+			input: Vec<u8>,
+		) -> DispatchResultWithPostInfo {
+			let who = ensure_signed(origin)?;
+
+			let caller_evm_address = T::EvmAccounts::evm_address(&who);
+			let context = CallContext::new_call(target, caller_evm_address);
+
+			let (exit_reason, output) = T::Evm::call(context, input.clone(), U256::zero(), T::GasLimit::get());
```

Comment:
please wrap `pallet_evm` call instead https://apidocs.bsx.fi/Hydration/pallets/EVM/extrinsic/call

with all of its parameters, instead of replicating parameters in the function, you should use call itself as the parameter